acados/ocp_nlp/ocp_nlp_common.h

```diff
@@ -247,6 +251,137 @@ typedef struct {
     // TODO(nielsvd): add cost and nonlinear constraints
 } ocp_nlp_solver_fcn_ptrs;
 
+
+
+
+
+/************************************************
+* headers
+************************************************/
+
+/************************************************
+* dims
+************************************************/
+
+/* ocp_nlp */
+
+//
+int ocp_nlp_dims_calculate_size(int N);
+//
+ocp_nlp_dims *ocp_nlp_dims_assign(int N, void *raw_memory);
+//
+void ocp_nlp_dims_init(int *nx, int *nu, int *nbx, int *nbu, int *ng, int *nh, int *ns, void *ocp_nlp_cost_dims, ocp_nlp_dims *dims);
+
+/* ocp_nlp_cost_ls */
+
+//
+int ocp_nlp_cost_ls_dims_calculate_size(int N);
+//
+ocp_nlp_cost_ls_dims *ocp_nlp_cost_ls_dims_assign(int N, void *raw_memory);
+//
+void ocp_nlp_cost_ls_dims_init(int *nv, int *ny, ocp_nlp_cost_ls_dims *dims);
+
+/************************************************
+* cost
+************************************************/
+
+/* least squares */
+
+//
+int ocp_nlp_cost_ls_calculate_size(ocp_nlp_cost_ls_dims *dims);
+//
+ocp_nlp_cost_ls *ocp_nlp_cost_ls_assign(ocp_nlp_cost_ls_dims *dims, void *raw_memory);
+
+/************************************************
+* dynamics
+************************************************/
+
+/* ERK */
+
+//
+int ocp_nlp_dynamics_erk_calculate_size(ocp_nlp_dims *dims);
+//
+ocp_nlp_dynamics_erk *ocp_nlp_dynamics_erk_assign(ocp_nlp_dims *dims, void *raw_memory);
+//
+void ocp_nlp_dynamics_erk_to_sim_in(ocp_nlp_dynamics_erk *dynamics, sim_in **sim);
```

Comment:
Module specific, doesn't belong in ``ocp_nlp_common.h``. Same applies to several other functions below.